facebook/__init__.py

```diff
@@ -200,18 +200,24 @@ def delete_request(self, user_id, request_id):
 
     def put_photo(self, image, album_path="me/photos", **kwargs):
         """
-        Upload an image using multipart/form-data.
+        Upload an image using image URL or multipart/form-data.
 
-        image - A file object representing the image to be uploaded.
+        image - An image URL or a file object representing the image to be uploaded.
         album_path - A path representing where the image should be uploaded.
-
         """
-        return self.request(
-            "{0}/{1}".format(self.version, album_path),
-            post_args=kwargs,
-            files={"source": image},
-            method="POST")
+        payload = {
+            'path': self.version + "/" + album_path,
+            'post_args': kwargs,
+            'method': "POST"
+        }
+
+        if isinstance(image, str):
+            kwargs['url'] = image
```

Comment:
Should this be `payload["url"]`? `kwargs` isn't part of `payload` which is the only object sent to the request method.

Reply:
payload['url'] does not work, I've try :( when you upload a file, its will be in files of request method and when you want an URL, it will be in POST params. In additional,  2 params is "source" for image www/enctyped encoded and "url" for Image Url so I keep "image" for being compatible with library